api/src/main/java/io/opentelemetry/trace/BigendianEncoding.java

```diff
@@ -165,5 +174,24 @@ private static void byteToBase16(byte value, char[] dest, int destOffset) {
     dest[destOffset + 1] = ENCODING[b | 0x100];
   }
 
+  public static boolean isValidBase16String(CharSequence value) {
+    for (int i = 0; i < value.length(); i++) {
+      char b = value.charAt(i);
+      // 48..57 && 97..102 are valid
+      if (!Character.isDigit(b) && (b < 97 || b > 102)) {
```

Comment:
It might affect other places as well, but while we're here, let's use a non-unicode check for `isDigit`

Reply:
happy to fix it. Are there non-ascii digits out there?

Reply:
done